### PR TITLE
add a new extension intel/SPV_INTEL_fpga_argument_interfaces

### PR DESCRIPTION
**Pull request by broxigarchen**
Added a new INTEL FPGA extension for scalar pointer decorations

### CODE REVIEW

**Review by bashbaug — CHANGES_REQUESTED** [review]
Hi, can you please:
- Include an HTML rendering for this extension.  I can help with this if needed.
- Update the README to include a link to the HTML rendering?

**Comment by broxigarchen** [comment]
> Hi, can you please:
> 
> * Include an HTML rendering for this extension.  I can help with this if needed.
> * Update the README to include a link to the HTML rendering?

done

**Review by bashbaug — COMMENTED** [review]
Thanks, latest changes LGTM!  Needs a non-Intel review approval before merging.

File: extensions/INTEL/SPV_INTEL_fpga_argument_interfaces.asciidoc
Comment (johnkslang) [inline extensions/INTEL/SPV_INTEL_fpga_argument_interfaces.asciidoc:148]: Minor misspelling, piplined -> pipelined.

File: extensions/INTEL/SPV_INTEL_fpga_argument_interfaces.asciidoc
Comment (bashbaug) [inline extensions/INTEL/SPV_INTEL_fpga_argument_interfaces.asciidoc:168]: Mentioned in last week's SPIR-V call: this condition is already described generally in the main SPIR-V specification so it does not need to be duplicated here.  From [3.20 Decoration](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_decoration):

> It is invalid to decorate any given id or structure member more than one time with the same decoration, unless explicitly allowed below for a specific decoration.

It would be best to remove this validation rule to eliminate the duplication.

It's fine to keep the following validation rule ("it is invalid to specify both ConduitKernelArgumentINTEL and RegisterMapKernelArgumentINTEL decorations on the same OpFunctionParameter") because it isn't already covered by the main SPIR-V specification.

Thanks!

Comment (broxigarchen) [reply extensions/INTEL/SPV_INTEL_fpga_argument_interfaces.asciidoc]: done

File: extensions/INTEL/SPV_INTEL_fpga_argument_interfaces.asciidoc
Comment (broxigarchen) [reply extensions/INTEL/SPV_INTEL_fpga_argument_interfaces.asciidoc]: done

**Comment by bashbaug** [comment]
Actually, one more minor issue: can you please build a new HTML spec with these updates?

**Pull request merged**
